pkg/system/system.go

```diff
@@ -401,6 +402,70 @@ func RunYaml(cmd *cobra.Command, args []string) {
 	util.Panic(p.PrintObj(sys, os.Stdout))
 }
 
+// CheckNooBaaImages runs a CLI command
+func CheckNooBaaImages(cmd *cobra.Command, sys *nbv1.NooBaa, args []string) string {
+	log := util.Logger()
+
+	noobaaImage := *sys.Spec.Image
+	noobaaActualImage := sys.Status.ActualImage
+	noobaaCoreStatefulset := util.KubeObject(bundle.File_deploy_internal_statefulset_core_yaml).(*appsv1.StatefulSet)
+	noobaaCoreStatefulset.Namespace = options.Namespace
+	util.KubeCheckQuiet(noobaaCoreStatefulset)
+	noobaaCoreStatefulsetImage := noobaaCoreStatefulset.Spec.Template.Spec.Containers[0].Image
+	if noobaaActualImage != noobaaCoreStatefulsetImage {
+		log.Warnf("⚠️  The image and actual image are not the same")
+	}
+	return noobaaImage
+}
+
+// CheckNooBaaDBImages runs a CLI command
+func CheckNooBaaDBImages(cmd *cobra.Command, sys *nbv1.NooBaa, args []string) string {
+	log := util.Logger()
+
+	noobaaDbImage := *sys.Spec.DBImage
+	noobaaDBStatefulset := util.KubeObject(bundle.File_deploy_internal_statefulset_db_yaml).(*appsv1.StatefulSet)
+	noobaaDBStatefulset.Namespace = options.Namespace
+	util.KubeCheckQuiet(noobaaDBStatefulset)
+	noobaaDBStatefulsetImage := noobaaDBStatefulset.Spec.Template.Spec.Containers[0].Image
+	if noobaaDbImage != noobaaDBStatefulsetImage {
```

Comment:
I think you will need to restrict this only when `sys.Spec.DBImage != nil` because if the spec did not specify the dbImage and the operator applied it's default image we don't want to fail or warn. Same for the running image - need to check if the statefulset even exists to avoid failing -
```suggestion
	desiredImage := ""
	runningImage := ""
	if sys.Spec.DBImage != nil {
		desiredImage = *sys.Spec.DBImage
	}
	sts := util.KubeObject(bundle.File_deploy_internal_statefulset_db_yaml).(*appsv1.StatefulSet)
	sts.Namespace = options.Namespace
	if util.KubeCheckQuiet(sts) {
		runningImage = sts.Spec.Template.Spec.Containers[0].Image
	}
	if desiredImage != "" && runningImage != "" && desiredImage != runningImage {
```